### DISCUSSION TITLE
How to insert an A tag with herf attribute?

### DISCUSSION THREAD

**Discussion by zhpgit** [ueberdosis/tiptap#1834] [General Questions]
Replace text when I want to add a link.

**Comment by hanspagel** [comment]
https://www.tiptap.dev/api/commands/insert-content

**Comment by zhpgit** [comment]
Thank you for reply：
&nbsp; &nbsp;Insert -content This can't insert a tag, any tag other than a can.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ueberdosis/tiptap"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月2日(星期四) 晚上8:19
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ueberdosis/tiptap] How to insert an A tag with herf attribute? (#1834)





 
https://www.tiptap.dev/api/commands/insert-content
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by duhaime** [reply]
Hmm, what version of tiptap are you on @zhpgit? I was able to insert a link just fine with the insertContent method:

```
import React, { useState } from 'react'
import { useEditor, EditorContent } from '@tiptap/react'
import StarterKit from '@tiptap/starter-kit'
import Link from '@tiptap/extension-link'

export default () => {
  const [markdownOutput, setMarkdownOutput] = useState("");

  const editor = useEditor({
    extensions: [
      StarterKit,
      Link,
    ],
    content: '',
  })

  return (
    <div>
      <MenuBar editor={editor} />
      <EditorContent editor={editor} />
    </div>
  )
}

const MenuBar = ({ editor }) => {
  if (!editor) {
    return null
  }

  return (
    <>
      <button onClick={() => {
        const url = 'https://google.com' || window.prompt('URL');
        const text = 'cool link' || window.prompt('text');
        editor.commands.insertContent(`<a href="${url}">${text}</a>`);
      }}>
        add link
      </button>
    </>
  )
}
```

using 

```
"@tiptap/extension-link": "^2.0.0-beta.35",
"@tiptap/react": "^2.0.0-beta.70",
"@tiptap/starter-kit": "^2.0.0-beta.117",
"react": "^17.0.2",
